presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -360,9 +362,10 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                                     .map(NodeRef::of)
                                     .map(columnReferences::get)
                                     .filter(Objects::nonNull)
+                                    .flatMap(List::stream)
                                     .collect(toImmutableList());
                             for (Expression sortKey : sortKeys) {
-                                if (!node.getArguments().contains(sortKey) && !fieldIds.contains(columnReferences.get(NodeRef.of(sortKey)))) {
+                                if (!node.getArguments().contains(sortKey) && !fieldIds.containsAll(columnReferences.get(NodeRef.of(sortKey)))) {
```

Comment:
Doesn't this cause problem if I have
SELECT ...  
FROM T1
JOIN T2 
  USING(x)
ORDER BY x

Add a test case for this.

Reply:
I'm not sure I follow, what is the problem? This is analyzing ORDER BY of aggregation expression, not the whole query.

Reply:
> I'm not sure I follow, what is the problem? This is analyzing ORDER BY of aggregation expression, not the whole query.

Since you are touching in the ORDER BY clause, I think it's good have the test as I suggested.